lib/utils.js

```diff
@@ -45,4 +45,28 @@ utils.removeHopToHopHeaders = function removeHopToHopHeaders(rh, removeContentEn
     });
 };
 
+/**
+ * From a list of uri Regex and values, constructs a regex to check if the
+ * request URI is in the white-list.
+ */
+var CACHE = new Map();
+utils.constructRegex = function(variants) {
+    if (CACHE.has(variants)) {
```

Comment:
Since variants is an array, it will use object equality for the match. This means that this cache won't necessarily be very effective.

I can think of two other options:

1) Cache compiled regexps (with a private property) on the options object passed to the filter. This should be the most efficient method.
2) Use stable-stringify to turn the array into a string key.